src/rtc.c

```diff
@@ -0,0 +1,792 @@
+ * 
+ * Internally, RTC cannot represent dates before 1990-01-01, although some
+ * RTC implementations (like UltraPIF) only support dates after 2000-01-01.
+ * 
+ * For highest compatibility, it is not recommended to set the date past
+ * 2038-01-19 03:14:07 UTC, which is the UNIX timestamp Epochalypse.
+ * 
+ * @param[in,out] rtc_time
+ *                Pointer to the RTC time data structure 
+ */
+void rtc_normalize_time( rtc_time_t * rtc_time )
+{
+    /* Clamp date/time values that have static limits */
+    if( rtc_time->year < 1900 ) rtc_time->year = 1900;
+    if( rtc_time->month > 11 ) rtc_time->month = 11;
+    if( rtc_time->hour > 23 ) rtc_time->hour = 23;
+    if( rtc_time->min > 59 ) rtc_time->min = 59;
+    if( rtc_time->sec > 59 ) rtc_time->sec = 59;
+
+    /* Clamp the day of the month based on the year and month */
+    uint16_t year = rtc_time->year;
+    uint8_t month = rtc_time->month;
+    uint8_t days_in_month = DAYS_IN_MONTH[month];
+    bool leap_year = year % 400 == 0 || (year % 100 != 0 && year % 4 == 0);
+    if( month == 1 && leap_year ) days_in_month = 29;
+    if( rtc_time->day > days_in_month ) rtc_time->day = days_in_month;
+
+    /* Recalculate the day of the week based on calendar date */
+    month = month + 1; /* Adjust month to 1-indexed */
+    if( month < 3 )
+    {
+        month = month + 12;
+        year = year - 1;
+    }
+    rtc_time->week_day = (
+        rtc_time->day
+        + (2 * month)
+        + (6 * (month + 1) / 10)
+        + year
+        + (year / 4)
+        - (year / 100)
+        + (year / 400)
+        + 1
+    ) % 7;
+}
+
+/**
+ * @brief Read the current date/time from the real-time clock.
+ * 
+ * If the RTC is not detected or supported, this function will
+ * not modify the destination rtc_time parameter.
+ *
+ * Your code should call this once per frame to update the #rtc_time_t
+ * data structure. The RTC Subsystem maintains a cache of the
+ * most-recent RTC time that was read and will only perform an
+ * actual RTC read command if the cache is invalidated. The
+ * destination rtc_time parameter will be updated regardless of
+ * the cache validity.
+ * 
+ * Cache will invalidate every #RTC_GET_CACHE_INVALIDATE_TICKS.
+ * Cache will also invalidate when the tick counter overflows.
+ * Calling #rtc_set will also invalidate the cache.
+ *
+ * @param[out]  rtc_time
+ *              Destination pointer for the RTC time data structure
+ * 
+ * @return whether the rtc_time destination pointer data was modified
+ */
+bool rtc_get( rtc_time_t * rtc_time )
+{
+    /* libdragon currently only supports getting the time for Joybus RTC! */
+    if( rtc_present() != RTC_JOYBUS ) return false;
+
+    /* This should be overwritten the first time this function is called */
+    static rtc_time_t cache_time = { 2000, 0, 1, 0, 0, 0, 6 };
+
+    uint32_t current_ticks = get_ticks();
+    int32_t distance = TICKS_DISTANCE( current_ticks, rtc_get_cache_ticks );
```

Comment:
The prototype is `TICKS_DISTANCE(from, to)`, so I think your usage here is reversed?

Reply:
Fixed by using `timer_ticks` instead of `get_ticks`